Library/Homebrew/test/requirement_spec.rb

```diff
@@ -59,17 +59,6 @@
     describe "#fatal is omitted" do
       it { is_expected.not_to be_fatal }
     end
-
-    describe "in subclasses" do
-      it "raises an error when instantiated" do
-        expect do
-          Class.new(described_class) do
-            def fatal? = false
-          end
-        end
-        .to raise_error(RuntimeError, /\AThe method `fatal\?` on #{described_class.name} was declared as final/)
```

Comment:
It's interesting that I added this test before I discovered that `:final` checks had to be explicitly [enabled](https://github.com/Homebrew/brew/pull/18801)

So it seems Sorbet is aware of being run in tests, and enables `:final` in that context, which was counterproductive in our case…